src/main/java/com/walshydev/streamdeck4j/Plugin.java

```diff
@@ -0,0 +1,133 @@
+package com.walshydev.streamdeck4j;
+
+import com.google.gson.JsonObject;
+import com.walshydev.streamdeck4j.hooks.EventListener;
+import com.walshydev.streamdeck4j.info.Application;
+import com.walshydev.streamdeck4j.info.Destination;
+import com.walshydev.streamdeck4j.info.Device;
+
+import javax.annotation.Nonnull;
+import java.awt.image.BufferedImage;
+import java.net.URL;
+import java.util.Set;
+
+@SuppressWarnings({"unused", "WeakerAccess", "SameParameterValue"})
+public interface Plugin {
+
+    /**
+     * Adds an event listener to the plugin
+     *
+     * @param eventListener The listener to add
+     */
+    void addListener(@Nonnull EventListener eventListener);
+
+    /**
+     * Get the {@link Application} info, this contains the language, platform and version being used.
+     *
+     * @return {@link Application} containing application info.
+     */
+    Application getApplication();
+
+    /**
+     * A collection of the {@link Device}s being used.
+     *
+     * @return A collection of the {@link Device}s being used.
+     */
+    Set<Device> getDevices();
+
+    /**
+     * Opens a URL on the PC the Stream Deck is connected to.
+     *
+     * @param url The URL to open
+     */
+    void openURL(@Nonnull URL url);
+
+    /**
+     * Sets the title of a specific button on the Stream Deck
+     *
+     * @param context     The unique identifier for the button
+     * @param title       The new title for the button
+     * @param destination The destination for the event (Hardware, Software or Both)
```

Comment:
What is this, old Destination docs? Please fix, absolutely unusable.

Reply:
Isn't that your job? ;)

Reply:
I didn't write these files ;)

Reply:
You wrote the documentation for them, that's enough ;)